app/controllers/clients_controller.rb

```diff
@@ -1,71 +1,57 @@
-    collection = filter_by_query params[:query], collection if params[:query].present?
-
-    collection = filter_by_symbol params[:id], collection if params[:id].present?
-    collection = filter_by_provider params[:provider_id], collection if params[:provider_id].present?
-    collection = filter_by_prefix params[:prefix], collection if params[:prefix].present?
-    collection = filter_by_year params[:year], collection if params[:year].present?
- 
-    collection = filter_by_ids params[:ids], collection if params[:ids].present?
- 
-    collection = Client.all if collection.respond_to?(:search)
-    # regions    = facet_by_region params, collection
-    years      = facet_by_year params, collection
-
-    page = params[:page] || {}
-    page[:number] = page[:number] && page[:number].to_i > 0 ? page[:number].to_i : 1
-    page[:size] = page[:size] && (1..1000).include?(page[:size].to_i) ? page[:size].to_i : 25
-    total = collection.count
-    providers = get_providers collection
-    
-    variable = case params[:sort]
-      when "created" then "created"
-      else "name"
+    if params[:id].present?
+      response = Client.find(params[:id])
+    elsif params[:ids].present?
+      response = Client.query(params[:ids])
+    else
+      page = (params.dig(:page, :number) || 1).to_i
+      size = (params.dig(:page, :size) || 25).to_i
+      from = (page - 1) * size
+
+      sort = case params[:sort]
+             when "name" then "name"
+             when "-name" then "name: desc"
+             when "created" then "created"
+             else "created"
+             end
+
+      params[:query] ||= "*"
+      response = Client.query(params[:query], from: from, size: size, sort: sort)
+      total = response.total
+      total_pages = (total.to_f / size).ceil
+
+      years = facet_by_year(response.response.aggregations.years.buckets)
+      providers = facet_by_provider(response.response.aggregations.providers.buckets)
     end
 
-    ordered = case params[:sort]
-        when "-name" then collection.sort_by { |hsh| hsh[variable] }.reverse
-        when "-created" then collection.sort_by { |hsh| hsh[variable] }.reverse
-        else collection.sort_by { |hsh| hsh[variable] }
-    end
+    @clients = Kaminari.paginate_array(response.results, total_count: total).page(page).per(size)
+
+    meta = {
+      total: total,
+      total_pages: total_pages,
+      page: page,
+      years: years
+    }
 
-      # collection.last.symbol.ddd
-    # https://github.com/elastic/elasticsearch-rails/issues/338
-    @clients = collection.all unless collection.respond_to?(:each)
-    @clients = Kaminari.paginate_array(ordered, total_count: total).page(page[:number]).per(page[:size])
-
-    meta = { total: total,
-             total_pages: @clients.total_pages,
-             page: page[:number].to_i,
-             providers: providers,
-             years: years 
-            }
-    
     render jsonapi: @clients, meta: meta, include: @include
```

Comment:
@mfenner Includes won't work unless the `Clients` have a `provider_id` . I think the new way in which you are importing from the API doesn't include provider_id and thus the include would not know how to add those ones. If you import from LUPO using the LUPO rake tasks then you `Clients` will come with `provider_ids` and `prefixes`